packages/yoastseo/src/languageProcessing/researches/getSubheadingTextLengths.js

```diff
@@ -18,12 +18,39 @@ export default function( paper, researcher ) {
 	// An optional custom helper to count length to use instead of countWords.
 	const customCountLength = researcher.getHelper( "customCountLength" );
 
-	const subHeadingTexts = [];
-	forEach( matches, function( subHeading ) {
-		subHeadingTexts.push( {
-			text: subHeading,
-			countLength: customCountLength ? customCountLength( subHeading ) : countWords( subHeading ),
+	const foundSubheadings = [];
+
+	forEach( matches, function( match ) {
+		foundSubheadings.push( {
+			subheading: match.subheading,
+			text: match.text,
+			countLength: customCountLength ? customCountLength( match.text ) : countWords( match.text ),
+			index: match.index,
 		} );
 	} );
-	return subHeadingTexts;
+
+	let textBeforeFirstSubheadingLength = 0;
+	let textBeforeFirstSubheading = "";
+	if ( foundSubheadings.length > 0 ) {
+		// Find first subheading.
+		const firstSubheading =  foundSubheadings[ 0 ];
+		// Retrieve text preceding first subheading.
+		textBeforeFirstSubheading = text.slice( 0, firstSubheading.index );
+		textBeforeFirstSubheadingLength = customCountLength
+			? customCountLength( textBeforeFirstSubheading )
+			: countWords( textBeforeFirstSubheading );
+	}
+
+	// Check if there is a text before the first subheading.
+	if ( textBeforeFirstSubheadingLength > 0 && textBeforeFirstSubheading !== "" ) {
+		// Also add the text before the first subheading to the array.
+		foundSubheadings.unshift( {
+			// Assign an empty string for the subheading for text that comes before the first subheading.
+			subheading: "",
+			text: textBeforeFirstSubheading,
+			countLength: textBeforeFirstSubheadingLength,
+		} );
```

Comment:
If there is a text before the first subheading, add the object to `foundSubheadings` array as the first element.